src/core/lwt_queue.mli

```diff
@@ -0,0 +1,42 @@
+(* Lightweight thread library for OCaml
+ * http://www.ocsigen.org/lwt
+ * Module Lwt_stream
+ * Copyright (C) 2014 Simon Cruanes
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, with linking exceptions;
+ * either version 2.1 of the License, or (at your option) any later
+ * version. See COPYING file for details.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *)
+
+(** {1 Blocking queue}
+This queue can be used for message passing between threads *)
+
+type 'a t
+(** Queue containing elements of type 'a *)
+
+val create : unit -> 'a t
+(** New queue with unbounded capacity *)
+
+val create_bounded : int -> 'a t
```

Comment:
Should the bound be an optional argument rather than a separate function?

    val create : ?max:int -> unit -> 'a t